aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -18,19 +20,19 @@ pub(crate) struct AptosExecutorTask<'a, S> {
 }
 
 impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
-    type Argument = &'a S;
+    type Argument = (&'a S, bool);
     type Error = VMStatus;
     type Output = AptosTransactionOutput;
     type Txn = SignatureVerifiedTransaction;
 
-    fn init(argument: &'a S) -> Self {
-        // AptosVM has to be initialized using configs from storage.
-        let vm = AptosVM::new_from_state_view(&argument);
-
-        Self {
-            vm,
-            base_view: argument,
+    fn init(argument: (&'a S, bool)) -> Self {
+        let (base_view, is_simulation) = argument;
+        let mut vm = AptosVM::new(&base_view.as_move_resolver());
+        if is_simulation {
+            vm = vm.for_simulation();
```

Comment:
Do we need this logic? Simulation api will only be invoked in a single thread setup and don't need the complication from block stm?

Reply:
The goal is to have simulation be 1:1 identical to execution. For example, for aggregators we are replacing Move values with identifiers at runtime, and then swap them back at commit time for fast processing. Converging simulation and execution makes this and other development less error-prone.

But I agree that BlockSTM set-up can be an overkill here, I was also thinking that in future we can simulate > 1 transaction.

Reply:
simulation won't be 1:1 identical because of the prologue checks? Though asides from that they should be identical. My take is that we should just converge the api for executing one single transaction but not try to converge the multi transaction api.

Reply:
I think only multisig prologue checks are different in simulation? Other types of transactions seems to follow the same path, if I am not mistaken

Reply:
It would be great to have it close to the real execution.  Today, those might be the only course, but if we add other transactions they may have different prologues.